techdocs/script/cli.py

```diff
@@ -21,17 +22,27 @@
 
     if args.command == "copy":
         try:
-            Copier(
-                Detector(
-                    args.from_path, args.to_path, ConfigLoader.default(args.to_path, fs).load(
-                        args.config_path
-                    )
-                ),
-                fs,
-                PrintingExecutor() if args.dry_run else Executor(fs),
-            ).execute(
-                args.from_path
-            )
+            # zz- prefix to make it the last file in the directory listing for example during PR review
+            with IndexLoader.loaded(
+                os.path.join(args.to_path, "zz-index"), fs, not args.dry_run
+            ) as index:
+                Copier(
+                    DeleteDetector(
+                        args.repo,
+                        index,
+                        args.from_path,
+                        args.to_path,
+                        CopyDetector(
+                            args.from_path,
+                            args.to_path,
+                            ConfigLoader.default(args.from_path, args.to_path, fs).load(
+                                args.config_path
+                            ),
+                        ),
```

Comment:
Is there any advantage to chaining those `Detectors` as decorators? 

Currently they sound more like `Operators` where we have:
- `DeleteOperator` - creates delete operations
- `CopyOperator` - creates copy operations

Why not go with `Copier(operators[],...)` and call them independently?

Reply:
I didn't like it, you'll have to pass a list (initially empty) to all the decorators, like:

```
all_operations = []
for detector in (First(), Second(), Third()):
  all_operations = detector.detect(filesystem, all_operations)
```

Reply:
On second thought, maybe it's more readable that way... I don't know if it is such big of a deal to refactor

Reply:
Not now, but if that diff case with Github would be another decorator on those I would think about it - not a blocking issue for me.